contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -566,9 +566,20 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
     function _sendTeleporterMessage(
         TeleporterMessageInput memory messageInput,
         TeleporterMessageReceipt[] memory receipts
-    ) private returns (uint256) {
+    ) private returns (bytes32) {
+        // If the blockchain ID has yet to be initialized, do so now.
+        bytes32 blockchainID_ = blockchainID;
+        if (blockchainID_ == bytes32(0)) {
+            blockchainID_ = WARP_MESSENGER.getBlockchainID();
+            blockchainID = blockchainID_;
+        }
+        require(blockchainID_ != bytes32(0), "TeleporterMessenger: blockchainID not set");
+
         // Get the message ID to use for this message.
-        uint256 messageID = _getNextMessageID(messageInput.destinationBlockchainID);
+        bytes32 messageID = _calculateNextMessageID();
+
+        // Increment the message nonce so the next message will have a different ID
+        ++messageNonce;
```

Comment:
It shouldn't matter functionally, but can we move this to after `bytes memory teleporterMessageBytes = abi.encode(teleporterMessage);`? 

This will be consistent with the pattern before, where we update the state variable pertaining to messageID after this line.

Additionally, I think it makes sense to update the `messageNonce` value only after it's used to contsruct the `TeleporterMessage` (even though now it's indirect)